www/src/theme/BlogPostPage/Metadata/index.tsx

```diff
@@ -0,0 +1,69 @@
+import { PageMetadata } from '@docusaurus/theme-common';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import { useBlogPost as ubp } from '@docusaurus/theme-common/internal';
+import { useBlogPost as typedUbp } from '@docusaurus/theme-common/lib/internal';
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
+import React from 'react';
+import { searchParams } from '../../../pages';
+import { getBaseUrl } from '../../getBaseUrl';
+
+// /lib doesn't actually export but actual export has import error in vscode
+const useBlogPost = ubp as unknown as typeof typedUbp;
+
+export default function BlogPostPageMetadata(): JSX.Element {
+  const { assets, metadata } = useBlogPost();
+  const { title, description, date, tags, authors, frontMatter } = metadata;
+  const { keywords } = frontMatter;
+  const _image = assets.image ?? frontMatter.image;
+  const {
+    siteConfig: { customFields },
+  } = useDocusaurusContext();
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const url = customFields!['url']! as string;
+
+  console.log({ metadata });
+
+  const author = authors[0];
+
+  const ogImg = `${getBaseUrl()}/api/ogDocs?${searchParams({
+    title: metadata.title,
+    authorName: author.name as string,
+    authorTitle: author.title as string,
+    authorImg: author.imageURL as string,
+    date,
+    readingTime: (metadata.readingTime as number).toString(),
+  })}`;
+
+  return (
+    <PageMetadata
+      title={title}
+      description={description}
+      keywords={keywords}
+      image={ogImg}
+    >
+      <meta property="og:type" content="article" />
+      <meta property={`og:title`} content={metadata.title} />
+      <meta property={`og:url`} content={url + metadata.permalink} />
```

Comment:
Maybe we can use the original component for all of this stuff and just add a `<Head><meta property={`og:url`} ....` as a child?

Something like

```ts

import { Metadata } from "@docusaurus/somewhere/metadata/";
import Head from '@docusaurus/Head';


export default function BlogPostPageMetadata(...props): JSX.Element {
  return (
    <Metadata {...props}>
     <Head><meta property={`og:url`} content={url + metadata.permalink} /></Head>
     {props.children}
   </Metadata>
  )
}
````